addons/uom/models/uom_uom.py

```diff
@@ -104,6 +104,15 @@ def create(self, vals_list):
 
     @api.multi
     def write(self, values):
+        if 'factor' in values or 'factor_inv' in values or 'category_id' in values or 'uom_type' in values:
+            try:
+                self.sudo().unlink()
```

Comment:
it is a strange method to test, but it is generic.
are you sure that rollback only the unlink ?

if not, just forbit to modifiy an unit. (an admin will have to update if case with SQL).
```python
def write(self, values):
     if 'factor' in values or 'factor_inv' in values or 'category_id' in values or 'uom_type' in values:
          raise UserError(_(
                    "You cannot change the ratio, the category or the type of this unit of mesure."
                    "If it is a new unit, delete it and create a new. Else archive and create a new."
                ))

def copy(self, default):
      raise UserError(_("It is not possible to duplicate an unit)")
```